website/docs/reference/global-configs/warnings.md

```diff
@@ -2,129 +2,216 @@
 title: "Warnings"
 id: "warnings"
 sidebar: "Warnings"
+toc_max_heading_level: 2
+intro_text: "dbt's warnings can be converted to errors using the --warn-error flag to promote all warnings to errors or --warn-error-options for granular control through options."
 ---
 
-Enabling `WARN_ERROR`config or setting the `--warn-error` flag will convert dbt warnings into errors. Any time dbt would normally warn, it will instead raise an error. Examples include `--select` criteria that selects no resources, deprecations, configurations with no associated models, invalid test configurations, or tests and freshness checks that are configured to return warnings.
+Enabling `WARN_ERROR` config or setting the `--warn-error` flag will convert _all_ dbt warnings into errors. Any time dbt would normally warn, it will instead raise an error. Examples include `--select` criteria that selects no resources, deprecations, configurations with no associated models, invalid test configurations, or tests and freshness checks that are configured to return warnings.
 
-You can use the `--warn-error` flag to promote all warnings to errors, such as:
-* Test warnings (for example, `LogTestResults`)
-* Jinja-level warnings (for example, `exceptions.warn`, or `JinjaLogWarning`)
-* Selection issues (for example, `NoNodesForSelectionCriteria`)
-* Adapter deprecation warnings (for example, `AdapterDeprecationWarning`)
+<VersionBlock firstVersion="1.8">
+
+:::caution Proceed with caution in production environments
+Using the `--warn-error` flag or `warn_error_options: error: "all"` will treat _all_ current and future warnings as errors.
+
+This means that if a new warning is introduced in a future version of dbt Core, your production job may start failing unexpectedly. We recommend proceeding with caution when doing this in production environments, and explicitly listing only the warnings you want to treat as errors in production.
+:::
+
+Instead of using the `--warn-error` flag to promote _all_ warnings to errors, you can use [`--warn-error-options`](#use---warn-error-options-for-targeted-warnings) flag to promote _specific_ warnings to errors, including:
```

Comment:
I wonder if this should actually be the start of the "Use `--warn-error-options` for targeted warnings" section